hyperspyui/mainwindowutillayer.py

```diff
@@ -336,6 +338,15 @@ def get_figure_filepath_suggestion(self, figure, deault_ext=None):
         # Analyze suggested filename
         base, tail = os.path.split(f)
         fn, ext = os.path.splitext(tail)
+        _logger.debug('fn before cleaning is: {}'.format(fn))
+
+        # Remove illegal characters and newlines from filename:
+        reserved_characters = '<>:"/\|?*'
```

Comment:
Do you need and `r'\'` here to make `\|` unambiguous?

Reply:
I don't believe so, because the `for c in...` treats the string as a list, so each character is processed individually:

```python
fn = "test_file_with<_and:_and/\\_and?_and\n_and|_and\""
reserved_characters = '<>:"/\|?*'
for c in reserved_characters:
    print(c)
    fn = fn.replace(c, '')
fn = fn.replace('\n', ' ')
fn
```
```
<
>
:
"
/
\
|
?
*
Out[17]: 'test_file_with_and_and_and_and _and_and'
```